Filenames with non-valid UTF8 in 'is' and 'nb' dictionaries prevent aspell from being installed

Hi,

I had aspell installed version 0.60.6.1. Doing a `brew update` today ended up in failure due to invalid UTF-8 characters in file names used by the 'is' and 'nb' dictionaries.

# Please always follow these steps:
- [ X ] Confirmed this is a problem with `brew install`ing one, specific Homebrew/homebrew-core formula (not cask or tap) and not every time you run `brew`? If it's a general `brew` problem please file this issue at https://github.com/Homebrew/brew/issues/new. If it's a `brew cask` problem please file this issue at https://github.com/caskroom/homebrew-cask/issues/new. If it's a tap (e.g. Homebrew/homebrew-php) problem please file this issue at the tap.
- [ X ] Ran `brew update` and retried your prior step?
- [ X ] Ran `brew doctor`, fixed all issues and retried your prior step?
- [  ] Ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
- [ X ] If `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

To help us debug your issue please explain:
- What you were trying to do (and why)

Trying to do a 'brew update' of everything. I do this every few weeks, to keep my system up to date.

- What happened (include command output)

The update failed on updating aspell, when doing the 'make install' for the 'is' dictionary. I commented that one out, and then it failed on the 'nb' dictionary too, for the same reason (different file name though). After also removing the 'nb' dictionary, then it finished correctly.

Gist with the build output and error:
   https://gist.github.com/lucrocha/198f61cb143f08724ee1b5f0aedc418c

Doing a `brew gist-logs aspell` failed for similar reason, invalid UTF-8 character in the output:
  https://gist.github.com/lucrocha/088ae16d265e0deba1a97f028ac532d9


Gist with the part of gist-logs for the 'is' install:
  https://gist.github.com/lucrocha/e436cff7a7cf5af1498e295bffdd952c

I've checked the contents of the 'is' package, https://ftp.gnu.org/gnu/aspell/dict/is/aspell-is-0.51.1-0.tar.bz2, as I expected the file to fail already when unpackaging it, but the file isn't named like that in the package:
-rw-r----- 1 lrocha wheel     72 Mar 11  2004 %EDslenska.alias

So the install does the conversion to an invalid character (it's the character `í` in latin1 encoding).

- What you expected to happen

That the install step of a package (or subpackage?) wouldn't try and fail to create invalid file names.

- Step-by-step reproduction instructions (by running `brew install` commands)

I have the option '--with-all-langs' enabled for aspell.
With a prior version installed with that option do: brew upgrade aspell

Without aspell installed, do: brew install aspell --with-all-langs

I can reproduce on my Sierra box. `brew install aspell --with-lang-nb` and `brew install aspell --with-lang-is` are 


```
$ brew install aspell --with-lang-nb
...
==> make install
Last 15 lines from /Users/janke/Library/Logs/Homebrew/aspell/12.make:
2017-11-03 15:26:33 -0400

make
install

/usr/local/Cellar/aspell/0.60.6.1_1/bin/word-list-compress d < nb.cwl | /usr/local/Cellar/aspell/0.60.6.1_1/bin/aspell  --lang=nb create master ./nb.rws
mkdir -p /usr/local/Cellar/aspell/0.60.6.1_1/lib/aspell-0.60/
cp nb.rws bokmal.alias bokm�l.alias nb.multi no.alias norsk.alias /usr/local/Cellar/aspell/0.60.6.1_1/lib/aspell-0.60/
cp: /usr/local/Cellar/aspell/0.60.6.1_1/lib/aspell-0.60/bokm�l.alias: Operation not permitted
make: *** [install] Error 1
...
$ brew install aspell --with-lang-is
...
==> make install
Last 15 lines from /Users/janke/Library/Logs/Homebrew/aspell/12.make:
2017-11-03 15:28:05 -0400

make
install

/usr/local/Cellar/aspell/0.60.6.1_1/bin/word-list-compress d < is.cwl | /usr/local/Cellar/aspell/0.60.6.1_1/bin/aspell  --lang=is create master ./is.rws
mkdir -p /usr/local/Cellar/aspell/0.60.6.1_1/lib/aspell-0.60/
cp is.rws icelandic.alias is.multi �slenska.alias /usr/local/Cellar/aspell/0.60.6.1_1/lib/aspell-0.60/
cp: /usr/local/Cellar/aspell/0.60.6.1_1/lib/aspell-0.60/�slenska.alias: Operation not permitted
make: *** [install] Error 1
```

I'll do a bit of research on this. I agree this looks like an encoding issue. On a quick glance, it looks like the `Makefile.pre` files in the aspell dictionary distributions are encoded in ISO-8859 instead of Unicode. But the build process/file system is operating in Unicode/UTF-8, so these are invalid strings. That is, I don't think it's the file names in the files themselves that are the problem; it's the file lists in this line of `Makefile.pre`.

```
multi_files = icelandic.alias is.multi íslenska.alias
```

Wonder why this wasn't reported earlier; the dictionary files haven't changed in years. Maybe nobody's using these languages?

--- a/Formula/aspell.rb
+++ b/Formula/aspell.rb
@@ -317,13 +317,6 @@ class Aspell < Formula
     sha256 "523912082848d891746dbb233f2ddb2cdbab6750dc76c38b3f6e000c9eb37308"
   end
 
-  option "with-lang-is", "Install is dictionary"
-  resource "is" do
-    url "https://ftp.gnu.org/gnu/aspell/dict/is/aspell-is-0.51.1-0.tar.bz2"
-    mirror "https://ftpmirror.gnu.org/aspell/dict/is/aspell-is-0.51.1-0.tar.bz2"
-    sha256 "3035bd29dad929ce66e6acdc7c29670df458e0d13fe178241b212f481111e3d6"
-  end
-
   option "with-lang-it", "Install it dictionary"
   resource "it" do
     url "https://ftp.gnu.org/gnu/aspell/dict/it/aspell6-it-2.2_20050523-0.tar.bz2"
@@ -427,13 +420,6 @@ class Aspell < Formula
     url "https://ftp.gnu.org/gnu/aspell/dict/mt/aspell-mt-0.50-0.tar.bz2"
     mirror "https://ftpmirror.gnu.org/aspell/dict/mt/aspell-mt-0.50-0.tar.bz2"
     sha256 "e00fcaad60a90cfed687ba02f62be8c27b8650457dd3c5bdcb064b476da059b4"
-  end
-
-  option "with-lang-nb", "Install nb dictionary"
-  resource "nb" do
-    url "https://ftp.gnu.org/gnu/aspell/dict/nb/aspell-nb-0.50.1-0.tar.bz2"
-    mirror "https://ftpmirror.gnu.org/aspell/dict/nb/aspell-nb-0.50.1-0.tar.bz2"
-    sha256 "e7746e8b617b2df576c1172399544818084524504202b16c747f52db5e5d228a"
   end
 
   option "with-lang-nds", "Install nds dictionary"
@@ -687,6 +673,15 @@ class Aspell < Formula
         system "make", "install"
       end
     end
+  end
+
+  def caveats
+    <<~EOS
+      Installation of the 'is' (Icelandic) and 'nb' (Norwegian) dictionaries is
+      currently broken. They can be installed manually.
+
+      See: https://github.com/Homebrew/homebrew-core/issues/28074
+    EOS
   end
 
   test do

